### PR TITLE
Changed to parsing using argparse

### PR DESCRIPTION
**Pull request by indocomsoft**

### CODE REVIEW

**Comment by indocomsoft** [comment]
This can help with adding further options such the `-d` option more sustainably.

Also I'm planning to add verbose `-v` option, since, I like showing implementation details.

**Comment by goweiwen** [comment]
This is a good idea.

I think actions shouldn't require hyphens, while flags should. For example, in `git commit --amend`, we don't use hyphens for the `commit` action, but use them for the `amend` flag.

**Comment by indocomsoft** [comment]
Done.

I also added some placeholder code for -d and -v when they are eventually implemented, for reference.

**Comment by goweiwen** [comment]
This is the current help message:
```
usage: ivle-sync [-h] {files,f,announcements,a,logout,l} ...

positional arguments:
  {files,f,announcements,a,logout,l}
                        Actions
    files (f)           Sync IVLE files to the current directory
    announcements (a)   Print out IVLE announcements
    logout (l)          Logout and clear token

optional arguments:
  -h, --help            show this help message and exit
Finished!
```
I find displaying the aliases redundant, but apparently, subparsers don't support the [allow_abrev](https://docs.python.org/dev/library/argparse.html#allow-abbrev) flag.

If we set it to like this instead (note the new `usage` and `title` arguments):
```py
        parser = argparse.ArgumentParser(usage="%(prog)s <action> [arguments]")
        subparsers = parser.add_subparsers(title="actions", dest='action')
```
we get this:
```
usage: ivle-sync <action> [arguments]

optional arguments:
  -h, --help            show this help message and exit

actions:
  {files,f,announcements,a,logout,l}
    files (f)           Sync IVLE files to the current directory
    announcements (a)   Print out IVLE announcements
    logout (l)          Logout and clear token
Finished!
```

We also want to get rid of the `Finished!` message when we're printing help. We can simply not catch the `SystemExit` exception and just print it after completing our actions (or remove it entirely, I'll leave it up to you).

If you'd like, you can extract the argument parsing into another function as well, as the `main` function is getting quite long.

Let me know what you think as this is all very opinionated.

**Comment by indocomsoft** [comment]
Personally, I think the aliases would be good for those who don't know those existed (me included before I read the source code).

But overall, I agree with what you have said. I've kept the "Finished!" message although that would probably belong better to the verbose mode. Keeping it around for when I implement the verbose mode.

I've implemented the changes, the code should be ready for merging.

File: ivle-sync.py
Comment (goweiwen) [inline ivle-sync.py]: This function should only parse arguments and return the `args` dict, and the if-else chain should be moved back into `main`.

File: ivle-sync.py
Comment (goweiwen) [inline ivle-sync.py]: This doesn't need to be in its own function. Eventually, we should use a logger with multiple severity levels instead of `print` to implement the verbose mode.

**Review by goweiwen — CHANGES_REQUESTED** [review]
Looks good! Just some minor refactoring before I merge it.

**Comment by indocomsoft** [comment]
Done

**Comment by goweiwen** [comment]
Good job! 👍

**Pull request merged**
